contrib/scripts/check_index_ovfps.sh

```diff
@@ -0,0 +1,526 @@
+        echo "                        2 : Avg_num_ovf_page_per_key"
+        echo "                        3 : Num_ovf_page"        
+	echo "  -t, --threshold=ARG Set the threshold to select the index to be reported; default: 1000"    
+	echo "  -s, --sort=ARG      Set sort filed name(pages, owner, table, index); default pages"
+        echo "  -v, --verbose       Set verbose mode; default disable" 
+	echo ""
+	echo " EXAMPLES"
+	echo "  $0 -S -o user1 demodb"
+	echo "  $0 -C -c game -t 100 demodb"
+	echo "  $0 -C -o user1 --key=1 --sort=table demodb"
+	echo ""
+	exit 1
+}
+
+function get_options()
+{
+	local sa_mode=0
+	local tsort=""
+
+	args=$(getopt -o :u:p:o:c:t:k:s:CSv  -l user:,password:,owner:,class:,threshold:,key:,sort:,--verbose   -- "$@" )
+	eval set -- "$args"
+	while true
+	do
+		case "$1" in
+			-C ) (( sa_mode++ ));  connection_mode="-C"  ;;
+			-S ) (( sa_mode++ ));  connection_mode="-S" ;;
+			-u | --user)  user="-u $2";       shift  ;;
+			-p | --password ) pass="-p $2";   shift ;;
+			-o | --owner) target_owner="$2";  shift ;;
+			-c | --class) target_table="$2";  shift ;;
+			-t | --threshold) threshold_value=$2; shift ;;
+			-k | --key)
+                                match_key_idx=$2
+                                if [ $2 -ge 1 ] && [ $2 -le  ${#key_titles[@]} ]; then
+                                   match_key="${key_titles[${match_key_idx} - 1]}"                                    
+                                else
+                                    show_usage
+				fi
+				shift ;;
+				
+			-s | --sort)
+				tsort=${2,,} # to lowercase
+				for (( i =0; i < ${#sort_title[@]}; i+=3 ))
+				do
+					if [ x"$tsort" == x"${sort_title[${i}]}" ] || [ x"$tsort" == x"${sort_title[${i} + 1]}" ]; then
+						sort_field=${sort_title[${i} + 2]}
+						break
+					fi
+				done
+				if [ $i -gt ${#sort_title[@]} ]; then
+					show_usage
+				fi
+				shift ;;
+                         -v | --verbose)
+                                verbose_mode=1 ;;                               
+			-- ) shift;  break ;;
+			* )  echo "unknown option"; show_usage ;;
+		esac
+		shift
+	done
+
+	if [ $# -ne 1 ]; then
+		show_usage
+	elif [ $sa_mode -gt 1 ]; then
+		show_usage
+	fi
+
+	database="$@"
+}
+
+function print_args()
+{
+#    echo arguments
+	echo_verbose "Args list..."
+	echo_verbose "connection_mode=${connection_mode}"
+	echo_verbose "user=${user}"
+	echo_verbose "pass=${pass}"
+	echo_verbose "owner=${target_owner}"
+	echo_verbose "class=${target_table}"
+	echo_verbose "key=${match_key_idx} (${match_key})"
+	echo_verbose "threshold=${threshold_value}"
+	echo_verbose "sort=${sort_title[${sort_field}-1*3]}"
+	echo_verbose "database=${database}"
+	echo_verbose "***************************************"
+	echo_verbose ""
+}
+
+function get_password ()
+{
+	local password
+
+	read -sp "Enter Password : " password
+	echo $password
+	echo "" > /dev/tty
+}
+
+function verify_user_pass ()
+{
+	local msg
+	local USERNAME
```

Comment:
Why did you capitalize only "username"?

Reply:
That's a quote from unloaddb.sh.
I'm guessing it was intended for clarity, since the dbuser name needs to be converted to uppercase and put in the SQL.